container-images/tcib/base/os/tempest/run_tempest.sh

```diff
@@ -58,13 +73,82 @@ if [[ ! -z ${TEMPESTCONF_REMOVE} ]]; then
     done <<< "$TEMPESTCONF_REMOVE"
 fi
 
-TEMPESTCONF_OVERRIDES="$(echo ${TEMPESTCONF_OVERRIDES} | tr '\n' ' ') identity.v3_endpoint_type public"
 
 if [ -n "$CONCURRENCY" ] && [ -z ${TEMPEST_CONCURRENCY} ]; then
     TEMPEST_ARGS+="--concurrency ${CONCURRENCY} "
 fi
 
-pushd $HOMEDIR
+function run_git_tempest {
+    mkdir -p $TEMPEST_EXTERNAL_PLUGIN_DIR
+    pushd $TEMPEST_EXTERNAL_PLUGIN_DIR
+
+    python3 -m venv .venv
+    source ./.venv/bin/activate
+
+    for plugin_index in "${!TEMPEST_EXTERNAL_PLUGIN_GIT_URL[@]}"; do
+        git_url=${TEMPEST_EXTERNAL_PLUGIN_GIT_URL[plugin_index]}
+        change_url=${TEMPEST_EXTERNAL_PLUGIN_CHANGE_URL[plugin_index]}
+        refspec=${TEMPEST_EXTERNAL_PLUGIN_REFSPEC[plugin_index]}
+        plugin_name=$(basename -s .git $git_url)
+
+        git clone $git_url
+
+        if [[ ! -z $change_url ]] && [[ ! -z $refspec ]] || \
+           [[ $change_url != "-" ]] && [[ $refspec != "-" ]]; then
+            pushd $plugin_name
+            git fetch $change_url $refspec
+            git checkout FETCH_HEAD
+            popd
+        fi
+
+        pip install ./$plugin_name
+    done
+
+    pushd $HOMEDIR
+    tempest init openshift
+    pushd $TEMPEST_DIR
+
+    discover-tempest-config ${TEMPESTCONF_ARGS} ${TEMPESTCONF_OVERRIDES}
+    tempest run ${TEMPEST_ARGS}
```

Comment:
hmm, what tempest is used here? the one installed from rpm? what if we wanna install tempest from source (e.g. to test a new patch etc). I would assume that if we pass tempest source as another plugin, tempest installed in the venv would override the globally installed one, so it should work. Any chance you tested that so that we're sure?

Reply:
hmm, what tempest is used here? the one installed from rpm?

- Either tempest that is installed as a dependency for a plugin
- or a tempest that is installed separately because a user specified it using the newly introduced variables in this patch.

Any chance you tested that so that we're sure?
- I hadn't tested it before I read this comment. I think you made a good point. But I've just tested it and the feature seems to be working even when a user specifies tempest via the newly introduced variables.

Reply:
thanks